pages/projects/[id]/edit.js

```diff
@@ -0,0 +1,58 @@
+import { Router, useRouter } from "next/router";
+import Form from "@/components/Form";
+import { useEffect, useState } from "react";
+import styled from "styled-components";
+
+export default function Edit({ projects, setNewProjects }) {
```

Comment:
You should have a helper function in your `_app.js` that is being passed down to this page and later called in your `editFunction`

```suggestion
export default function Edit({ projects, onUpdateProject }) {
```